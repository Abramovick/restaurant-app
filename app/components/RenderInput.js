import React from 'react';
import PropTypes from 'prop-types';
import { Input, Item, Icon } from 'native-base';
import { HomeScreenStyles as styles } from '../styles/styles';

const RenderInput = ({
  placeholder,
  meta: { touched, error },
  input: { onChange, ...restInput }
}) =>
    <Item style={(touched && error) ?
      [styles.inputView, styles.errorStyle] : styles.inputView}>
      {(touched && error) && <Icon style={styles.searchIcon} name='alert'/>}
      <Input
      style={styles.input}
      onChangeText={onChange}
      {...restInput}
      placeholder={placeholder}
      placeholderTextColor="white"/>
    </Item>;

RenderInput.propTypes = {
  placeholder: PropTypes.string,
};

RenderInput.propTypes = {
  meta: PropTypes.object,
  input: PropTypes.object,
  placeholder: PropTypes.string,
  onChangeText: PropTypes.func,
};

export default RenderInput;
