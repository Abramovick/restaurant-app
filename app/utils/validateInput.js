export const validate = (values) => {
  const errors = {};
  if (!values.firstName) {
    errors.firstName = 'Required';
  } else if (values.firstName.length > 15) {
    errors.firstName = 'Must be 15 characters or less';
  }
  if (!values.lastName) {
    errors.lastName = 'Required';
  } else if (values.lastName.length > 15) {
    errors.lastName = 'Must be 15 characters or less';
  }
  if (!values.email) {
    errors.email = 'Required';
  } else if (!/^.+@.+$/i.test(values.email)) {
    errors.email = 'Invalid email address';
  }
  if (!values.password) {
    errors.password = 'Password is required';
  } else if (values.password.length < 6) {
    errors.password = 'Password too short';
  }
  if (!values.passwordConfirm) {
    errors.passwordConfirm = 'Password Confirm is required';
  } else if (values.password !== values.passwordConfirm) {
    errors.passwordConfirm = 'Passwords do not match';
  }

  if (!values.phoneNum) {
    errors.phoneNum = 'Phone Number is required';
  }
  return errors;
};
