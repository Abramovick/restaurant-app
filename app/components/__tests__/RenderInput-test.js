import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import RenderInput from '../RenderInput';
import { validate } from '../../utils';

const meta = {
  touched: true,
  error: true
};
const input = {
  onChange: jest.fn()
};

it('renders the input', async () => {
  const tree = renderer.create(<RenderInput meta={meta} input={input}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the input', async () => {
  const tree = renderer.create(<validate />).toJSON();
  expect(tree).toMatchSnapshot();
});
