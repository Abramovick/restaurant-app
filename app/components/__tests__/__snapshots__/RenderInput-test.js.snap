// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the input 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E0956D",
      "borderBottomWidth": 1,
      "borderColor": "red",
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopWidth": 0,
      "flex": 1,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "center",
      "marginLeft": 2,
      "marginTop": "3%",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
  <TextInput
    allowFontScaling={true}
    editable={true}
    onChangeText={[Function]}
    placeholder={undefined}
    placeholderTextColor="white"
    style={
      Array [
        Object {
          "color": "#000",
          "flex": 1,
          "fontSize": 17,
          "height": 50,
          "lineHeight": 24,
          "paddingLeft": 5,
          "paddingRight": 5,
          "top": 1.5,
        },
        Object {
          "color": "#424242",
          "flex": 1,
          "fontFamily": "SinkinSans-200XLight",
          "fontSize": 12,
          "paddingBottom": 10,
          "paddingLeft": 0,
          "paddingRight": 10,
          "paddingTop": 10,
        },
      ]
    }
    underlineColorAndroid="rgba(0,0,0,0)"
    virtual={undefined}
  />
</View>
`;

exports[`renders the input 2`] = `<validate />`;
